source/modulo_components/include/modulo_components/Component.hpp

```diff
@@ -143,8 +144,9 @@ inline void Component::add_output(
         break;
       }
     }
-  } catch (const std::exception& ex) {
-    // TODO if modulo::communication had a base exception, could catch that
+  } catch (const modulo_core::exceptions::CoreException& ex) {
+    RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to add output '" << signal_name << "': " << ex.what());
```

Comment:
In these cases when there are identical branches, it discards the benefit of catching the custom exceptions. I would suggest just adding that additional info to the logger (since `ex.what()` does not actually tell us the exception type). This kind of thing might make debugging from logs easier in future.
```suggestion
    RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to add output '" << signal_name << "'! CoreException: " << ex.what());
```